src/unicode.cpp

```diff
@@ -101,7 +101,8 @@ namespace ada::unicode {
    * @see https://github.com/nodejs/node/blob/main/src/node_url.cc#L226
    */
   std::string utf8_percent_encode(const std::string_view input, const uint8_t character_set[]) noexcept {
-    std::string result;
+    std::string result{};
+    result.reserve(input.length());
 
     for (auto iterator: input) {
```

Comment:
The 'auto' in the following loop...

```C++
for (auto iterator: input) {
```

is a `char`. Under many systems, a `char` is a signed type which is maybe not what you want. I recommend the following...

```C++
for (unsigned char iterator: input) {
```

Or a more explicit version...

```C++
for (uint8_t iterator: input) {
```

Reply:
Thanks I updated the implementation